contracts/SSVNetwork.sol

```diff
@@ -747,38 +748,53 @@ contract SSVNetwork is UUPSUpgradeable, OwnableUpgradeable, ISSVNetwork {
     /**************************/
 
     function updateNetworkFee(uint256 fee) external override onlyOwner {
-        Network memory network_ = network;
+        uint64 newFee = fee.shrink();
+        
+        bytes32 fnData = keccak256(abi.encodeWithSelector(this.updateNetworkFee.selector, fee));      
+        if (!_isFunctionLocked(this.updateNetworkFee.selector, fnData)) {
+            Network memory network_ = network;
 
-        DAO memory dao_ = dao;
-        dao_ = dao_.updateDAOEarnings(network.networkFee);
-        dao = dao_;
+            DAO memory dao_ = dao;
+            dao_ = dao_.updateDAOEarnings(network_.networkFee);
+            dao = dao_;
 
-        network_.networkFeeIndex = NetworkLib.currentNetworkFeeIndex(network_);
-        network_.networkFeeIndexBlockNumber = uint64(block.number);
+            network_.networkFeeIndex = NetworkLib.currentNetworkFeeIndex(network_);
+            network_.networkFeeIndexBlockNumber = uint64(block.number);
 
-        emit NetworkFeeUpdated(network_.networkFee.expand(), fee);
+            emit NetworkFeeUpdated(network_.networkFee.expand(), fee);
 
-        network_.networkFee = fee.shrink();
-        network = network_;
+            network_.networkFee = newFee;
+            network = network_;
+            
+            delete _timelocks[fnData];
+        }
+        
     }
 
     function withdrawNetworkEarnings(
         uint256 amount
     ) external override onlyOwner {
-        DAO memory dao_ = dao;
-
         uint64 shrunkAmount = amount.shrink();
+        
+        bytes32 fnData = keccak256(
+            abi.encodeWithSelector(this.updateNetworkFee.selector, amount)
+        );      
+        if (!_isFunctionLocked(this.withdrawNetworkEarnings.selector, fnData)) {
+            DAO memory dao_ = dao;
 
-        if (shrunkAmount > dao_.networkBalance(network.networkFee)) {
-            revert InsufficientBalance();
-        }
+            if(shrunkAmount > dao_.networkBalance(network.networkFee)) {
```

Comment:
Please just double check that dao_.networkBalance(network.networkFee) is returning a shrunk amount not expanded.